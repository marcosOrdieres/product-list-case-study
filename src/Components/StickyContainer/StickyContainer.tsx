import styled from 'styled-components';

export interface StickyContainerProps {
  children?: JSX.Element[];
}

const StickyContainerStyled = styled.div`
  position: sticky;
  display: flex;
  flex-direction:column;
  align-items: center;
  top:0;
  z-index:1;
  background-color: white;
`;

export const StickyContainer: React.FC<StickyContainerProps> = (props) => <StickyContainerStyled {...props}>{props.children}</StickyContainerStyled>;

export default StickyContainerStyled;