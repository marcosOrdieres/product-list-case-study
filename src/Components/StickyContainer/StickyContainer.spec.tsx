import { render } from '@testing-library/react';
import React from 'react';

import StickyContainer from './StickyContainer';

describe('StickyContainer', () => {
  it('should render StickyContainer and match styles', () => {
    const { asFragment } = render(<StickyContainer>StickyContainer</StickyContainer>);
    expect(asFragment()).toMatchSnapshot();
  });
});
