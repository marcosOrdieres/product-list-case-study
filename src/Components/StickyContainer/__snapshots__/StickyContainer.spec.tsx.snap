// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StickyContainer should render StickyContainer and match styles 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBQB ceKeiK"
  >
    StickyContainer
  </div>
</DocumentFragment>
`;
